Lib/typing.py

```diff
@@ -2839,14 +2841,22 @@ def __new__(cls, name, bases, ns, total=True):
         Subclasses and instances of TypedDict return actual dictionaries.
         """
         for base in bases:
-            if type(base) is not _TypedDictMeta:
+            if type(base) is not _TypedDictMeta and base is not Generic:
                 raise TypeError('cannot inherit from both a TypedDict type '
                                 'and a non-TypedDict base class')
-        tp_dict = type.__new__(_TypedDictMeta, name, (dict,), ns)
+
+        if any(issubclass(b, Generic) for b in bases):
+            generic_base = (Generic,)
+        else:
+            generic_base = ()
+
+        tp_dict = type.__new__(_TypedDictMeta, name, (*generic_base, dict,), ns)
+        if generic_base:
+            class_getitem = Generic.__class_getitem__.__func__
+            tp_dict.__class_getitem__ = classmethod(class_getitem)
```

Comment:
It is not even needed, because `Generic` is before `dict` in `bases`.

Reply:
Fixed in my latest push